test/extended/builds/sti_secrets.go

```diff
@@ -0,0 +1,105 @@
+package builds
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"k8s.io/kubernetes/test/e2e"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("default: S2I incremental build with push and pull to authenticated registry", func() {
+	defer g.GinkgoRecover()
+	var (
+		templateFixture = exutil.FixturePath("fixtures", "incremental-auth-build.json")
+		oc              = exutil.NewCLI("build-sti-env", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Describe("Building from a template", func() {
+		g.It(fmt.Sprintf("should create a build from %q template and run it", templateFixture), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By(fmt.Sprintf("calling oc new-app -f %q", templateFixture))
+			err := oc.Run("new-app").Args("-f", templateFixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("starting a test build")
+			buildName, err := oc.Run("start-build").Args("initial-build").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting the build is in Complete phase")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName,
+				// The build passed
+				func(b *buildapi.Build) bool {
+					return b.Name == buildName && b.Status.Phase == buildapi.BuildPhaseComplete
+				},
+				// The build failed
+				func(b *buildapi.Build) bool {
+					if b.Name != buildName {
+						return false
+					}
+					return b.Status.Phase == buildapi.BuildPhaseFailed || b.Status.Phase == buildapi.BuildPhaseError
+				},
```

Comment:
yup...i copy+pasted w/o studying it too carefully, but this definitely seems like a spot for a utility func "waitforbuildsuccess" or some such thing.  will look at it tomorrow.